src/Microsoft.DotNet.Cli.Utils/MarkOfTheWebDetector.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.PlatformAbstractions;
+using RuntimeEnvironment = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class MarkOfTheWebDetector : IMarkOfTheWebDetector
+    {
+        private const string ZoneIdentifierStreamName = "Zone.Identifier";
+        private const string ZoneIdIs3String = "ZoneId=3";
+
+        public bool HasMarkOfTheWeb(string filePath)
+        {
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"{filePath} cannot be found");
+            }
+
+            if (RuntimeEnvironment.OperatingSystemPlatform != Platform.Windows)
+            {
+                return false;
+            }
+
+            return ZoneIdIs3(filePath);
+        }
+
+        private static bool ZoneIdIs3(string filePath)
+        {
+            return AlternateStream
+                            .ReadAlternateStream(filePath, ZoneIdentifierStreamName)
+                            .Split(new [] { Environment.NewLine }, StringSplitOptions.None)
+                            .Any(l => l.Equals(ZoneIdIs3String, StringComparison.Ordinal));
```

Comment:
It's wasteful to read the whole thing into one concatenated string, then split it by newlines. Could also have used ReadLine() instead of ReadToEnd() and checked each one.

I'm guessing this file is usually just one line, though?

Reply:
No, it is a INI file.  Like https://textplain.files.wordpress.com/2016/04/image.png?w=622&h=164&zoom=2

I am struggling with the 2 ends. Parse it as INI and match key is zone valve is 3 or just do a string contain of the whole file.

Maybe just string contain of the whole file is enough? there is no good ini parser.

Reply:
Maybe we can avoid parsing and borrow the code from msbuild...